src/cartesian/getTicks.ts

```diff
@@ -202,49 +120,27 @@ export function getTicks(props: CartesianAxisProps, fontSize?: string, letterSpa
 
   let candidates: CartesianTickItem[] = [];
 
+  const sizeKey = orientation === 'top' || orientation === 'bottom' ? 'width' : 'height';
+  const unitSize: Size =
+    unit && sizeKey === 'width' ? getStringSize(unit, { fontSize, letterSpacing }) : { width: 0, height: 0 };
+
+  const getTickSize = (content: CartesianTickItem, index: number) => {
+    return getSizeOfTick(tickFormatter, content, index, sizeKey, fontSize, letterSpacing, unitSize, angle);
```

Comment:
this feels slightly confusing, why wrap a function with a function? Concerned this reduces readability of this file when the goal is the opposite. The rest looks good (and I know this is needed for that), but maybe we can look at this again?

Reply:
It is called currying and basically partially applies a function. https://javascript.info/currying-partials

The idea is that now I only pass this function down, as opposed to passing down:
- tickFormatter
- sizeKey
- fontSize
- letterSpacing
- unitSize
- angle

Reply:
It ihmo improves readability downstream, where we only call `getStringSize(content, index)``

Reply:
Yep makes sense. Sorry was kind of hard to read in the code review format